src/components/Hooks/useFetchCountries.jsx

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect,useMemo } from 'react';
+import axios from 'axios';
+
+let countriesData = [];
+
+function useFetchCountries(selectedRegion, searchValue) {
+  const [countries, setCountries] = useState([]);
+
+  const fetchData = async () => {
+    try {
+      const { data } = await axios('https://restcountries.com/v3.1/all');
+      setCountries(data);
+      countriesData = data;
+    } catch (error) {
+      console.log("error")
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+
+  useEffect(() => {
+    let newCountries = {};
+    if (selectedRegion === 'All') newCountries = countriesData;
+    else {
+      newCountries = countriesData.filter(
+        (country) =>
```

Comment:
In here you can make this shorter bu doing something like this : 
```
if(someCondition){
  aVariable = bVariable
  return
}

//Rest of the code in here
```

This is called guard close and it helps us exit the function early